### PR TITLE
[GCC/4.8.2] Contrib a MacOSX-buildable Lmod/5.8.5 - and not only (OK)

### PR DESCRIPTION
**Pull request by fgeorgatos**
Get ready for Lmod/5.8, installable on recent macosx variants (implies a tiny patch)

### CODE REVIEW

**Comment by fgeorgatos** [comment]
@boegel: you asked it, you got it ;-) 

testing is under way, it should be ready for REVIEW in a few minutes,,,

**Comment by fgeorgatos** [comment]
Test report by @fgeorgatos
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
Linux debian 6.0.10, Intel(R) Xeon(R) CPU           X5675  @ 3.07GHz, Python 2.6.6
See https://gist.github.com/89f3720c34f2a05543a0 for a full test report.

File: easybuild/easyconfigs/n/ncurses/ncurses-5.9-GCC-4.8.2-no-cpp-patch.eb [hunk 1]
```diff
@@ -1,11 +1,16 @@
 name = 'ncurses'
 version = '5.9'
+versionsuffix = '-no-cpp-patch'
 
 homepage = 'http://www.gnu.org/software/ncurses/'
 description = """The Ncurses (new curses) library is a free software emulation of curses in System V Release 4.0,
  and more. It uses Terminfo format, supports pads and color and multiple highlights and forms characters and
  function-key mapping, and has all the other SYSV-curses enhancements over BSD Curses."""
 
+import platform
```
Comment (boegel) [inline easybuild/easyconfigs/n/ncurses/ncurses-5.9-GCC-4.8.2-no-cpp-patch.eb]: drop this

File: easybuild/easyconfigs/n/ncurses/ncurses-5.9-GCC-4.8.2-no-cpp-patch.eb [hunk 2]
```diff
@@ -1,11 +1,16 @@
...
+
+patches = ['configure_darwin.patch']
```
Comment (boegel) [inline easybuild/easyconfigs/n/ncurses/ncurses-5.9-GCC-4.8.2-no-cpp-patch.eb]: include the software name/version in the patch: `ncurses-5.9_configure_darwin.patch`

File: easybuild/easyconfigs/n/ncurses/ncurses-5.9-GCC-4.8.2-no-cpp-patch.eb [hunk 3]
```diff
@@ -1,11 +1,16 @@
 name = 'ncurses'
 version = '5.9'
+versionsuffix = '-no-cpp-patch'
```
Comment (boegel) [inline easybuild/easyconfigs/n/ncurses/ncurses-5.9-GCC-4.8.2-no-cpp-patch.eb]: drop the versionsuffix, you're not changing the behaviour of the software at all here, so no versionsuffix is needed

oh, and don't change the name for existing files, just create a copy (which you won't need anymore now)

File: easybuild/easyconfigs/n/ncurses/configure_darwin.patch [hunk 4]
```diff
@@ -0,0 +1,11 @@
+--- ncurses-5.9.orig/configure	2014-10-25 21:11:31.000000000 +0200
```
Comment (boegel) [inline easybuild/easyconfigs/n/ncurses/configure_darwin.patch]: add some info on who patched this, why the patch is required

File: easybuild/easyconfigs/l/Lua/Lua-5.1.4-8-GCC-4.8.2.eb [hunk 5]
```diff
@@ -16,7 +16,7 @@ toolchain = {'name': 'GCC', 'version': '4.8.2'}
 sources = ['lua-%s.tar.gz' % version.replace('-', '.')]
 source_urls = ['http://sourceforge.net/projects/lmod/files/']
 
-dependencies = [('ncurses', '5.9')]
+dependencies = [('ncurses', '5.9', '-no-cpp-patch')]
```
Comment (boegel) [inline easybuild/easyconfigs/l/Lua/Lua-5.1.4-8-GCC-4.8.2.eb]: drop version suffix

File: easybuild/easyconfigs/l/Lua/Lua-5.1.4-5-GCC-4.8.2.eb [hunk 6]
```diff
@@ -16,7 +16,7 @@ toolchain = {'name': 'GCC', 'version': '4.8.2'}
 sources = ['lua-%s.tar.gz' % version.replace('-', '.')]
 source_urls = ['http://sourceforge.net/projects/lmod/files/']
 
-dependencies = [('ncurses', '5.9')]
+dependencies = [('ncurses', '5.9', '-no-cpp-patch')]
```
Comment (boegel) [inline easybuild/easyconfigs/l/Lua/Lua-5.1.4-5-GCC-4.8.2.eb]: drop version suffix

File: easybuild/easyconfigs/l/Lmod/Lmod-5.8.4-GCC-4.8.2.eb [hunk 7]
```diff
@@ -0,0 +1,27 @@
+easyblock = 'ConfigureMake'
+
+name = "Lmod"
+version = "5.8.4"
```
Comment (boegel) [inline easybuild/easyconfigs/l/Lmod/Lmod-5.8.4-GCC-4.8.2.eb]: please bump to `5.8.5`

Comment (fgeorgatos) [inline easybuild/easyconfigs/l/Lmod/Lmod-5.8.4-GCC-4.8.2.eb]: @boegel: check!

File: easybuild/easyconfigs/n/ncurses/ncurses-5.9-GCC-4.8.2.eb [hunk 8]
```diff
@@ -6,6 +6,9 @@ description = """The Ncurses (new curses) library is a free software emulation o
  and more. It uses Terminfo format, supports pads and color and multiple highlights and forms characters and
  function-key mapping, and has all the other SYSV-curses enhancements over BSD Curses."""
 
+# in relation to -no-cpp-precomp bug: https://lists.gnu.org/archive/html/bug-ncurses/2012-12/msg00001.html
+patches = ['ncurses-5.9_configure_darwin.patch']
```
Comment (boegel) [inline easybuild/easyconfigs/n/ncurses/ncurses-5.9-GCC-4.8.2.eb]: @wpoely86: you may want to integrate this bit in #1337 too, while you're at it... :-)

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/a32651b4eff7f7258346 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (2 easyconfigs in this PR)
Darwin UNKNOWN UNKNOWN, Intel(R) Core(TM) i5-4258U CPU @ 2.40GHz, Python 2.7.6
See https://gist.github.com/db6a89ccb54de5b1af65 for a full test report.

**Comment by fgeorgatos** [comment]
Test report by @fgeorgatos
**SUCCESS**
Build succeeded for 2 out of 2
Linux debian 6.0.10, Intel(R) Xeon(R) CPU E5-2660 0 @ 2.20GHz, Python 2.6.6
See https://gist.github.com/a40ed8b79b5a22fffff7 for a full test report.

**Comment by boegel** [comment]
Looking good, so going in. Thanks @fgeorgatos!

**Pull request merged**